src/autodiff_nlsmodel.jl

```diff
@@ -61,3 +73,101 @@ function hprod_residual!(nls :: ADNLSModel, x :: AbstractVector, i :: Int, v ::
   Hiv[:] = ForwardDiff.hessian(x->nls.F(x)[i], x) * v
   return Hiv
 end
+
+function cons(nls :: ADNLSModel, x :: Vector)
+  increment!(nls, :neval_cons)
+  return nls.c(x)
+end
```

Comment:
If we asked the user to supply `cons!` instead of `cons`, would we loose efficiency in the AD computations?

Reply:
I don't think so. Do we change this is general, instead of only for NLS AD?

Reply:
The only other places would be SimpleNLPModel and ADNLPModels, no?

Reply:
Yes. It should be simple.

Reply:
I've checked if using only `c!(x, cx)` would be possible, but it doesn't appear to be because of the Hessian. We could have the user give both `c` and `c!`, but I'm not sure that's a good idea.

Reply:
I don't think that would be a good idea. What's the issue with the Hessian?

Reply:
I'm computing the Hessian of the i-th constraint as
```julia
ForwardDiff.hessian(x->nlp.c(x)[i], x)
```
and couldn't make it work using `c!`.

Reply:
That's odd because `cons!(x, c)` is supposed to return `c`, so I would have expected `x -> nlp.cons!(x, c)[i]` to work. But if it doesn't, oh well.

Reply:
It gives some conversion error with Dual